src/components/input/file/index.tsx

```diff
@@ -21,69 +33,130 @@ export class SmoothlyInputFile implements Clearable, Input {
 	@Prop({ reflect: true }) showLabel = true
 	@Prop({ mutable: true }) value?: File
 	@Prop({ mutable: true, reflect: true }) placeholder: string | undefined
+	@State() dragging = false
 	@Event() smoothlyInputLooks: EventEmitter<(looks: Looks, color: Color) => void>
-	@Event() smoothlyInput: EventEmitter<Record<string, File>>
-	@Event() smoothlyChange: EventEmitter<Record<string, File>>
+	@Event() smoothlyInput: EventEmitter<Record<string, any>>
+	@Event() smoothlyInputLoad: EventEmitter<(parent: HTMLElement) => void>
+	@Event() smoothlyFormDisable: EventEmitter<(disabled: boolean) => void>
+	private listener: { changed?: (parent: Editable) => Promise<void> }
+	private transfer: DataTransfer = new DataTransfer()
+	private input?: HTMLInputElement
+	private initialValue: SmoothlyInputFile["value"]
+
+	private get files(): FileList {
+		this.transfer.items.clear()
+		this.value && this.transfer.items.add(this.value)
+		return this.transfer.files
+	}
 
-	componentWillLoad() {
+	componentWillLoad(): void {
 		this.smoothlyInputLooks.emit((looks, color) => ((this.looks = looks), !this.color && (this.color = color)))
+		this.smoothlyInput.emit({ [this.name]: this.value })
+		this.smoothlyInputLoad.emit(() => {})
+		!this.readonly && this.smoothlyFormDisable.emit(readonly => (this.readonly = readonly))
 	}
 
 	@Method()
 	async clear(): Promise<void> {
 		this.value = undefined
 	}
-	render() {
+	@Method()
+	async listen(property: "changed", listener: (parent: Editable) => Promise<void>): Promise<void> {
+		this.listener[property] = listener
+		listener(this)
+	}
+	@Method()
+	async edit(editable: boolean): Promise<void> {
+		this.readonly = !editable
+	}
+	@Method()
+	async reset(): Promise<void> {
+		this.value = this.initialValue
+	}
+	@Method()
+	async setInitialValue(): Promise<void> {
+		this.initialValue = this.value
+		this.valueChanged()
+	}
+	@Method()
+	async binary(): Promise<boolean> {
+		return true
+	}
+
+	@Watch("value")
+	valueChanged(): void {
+		this.changed = this.initialValue !== this.value
+		this.smoothlyInput.emit({ [this.name]: this.value })
+	}
+
+	inputHandler(event: Event): void {
+		event.stopPropagation()
+		if (this.input?.files?.length) {
+			this.value = this.input?.files[0]
+			this.smoothlyInput.emit({ [this.name]: this.value })
```

Comment:
Do not emit smoothlyInput here, its enough for the value watcher to emit when value changes

Reply:
\+